### PR TITLE
Deprecate cachegroup parameters and remove from TO API v4 and TP

### PR DESCRIPTION
**Pull request by rawlinp**
## What does this PR (Pull Request) do?
This PR deprecates cachegroup parameters and removes them from TO API v4 and removes all associated functionality from TP.

## Which Traffic Control components are affected by this PR?
- Traffic Control Client (Go)
- Traffic Ops
- Traffic Portal

## What is the best way to verify this PR?
Verify that the removed TO API routes still work pre-4.0 but don't work at 4.0. Verify the TO API tests pass. Verify TP cachegroups pages still function.

## The following criteria are ALL met by this PR
- [x] deprecation, no new tests necessary
- [x] deprecation, no docs necessary
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

**Comment by rawlinp** [comment]
NOTE: TO API tests are still failing in GHA due to the DB locale sorting issue that is going to be fixed by https://github.com/apache/trafficcontrol/pull/5957

File: traffic_ops/testing/api/v4/traffic_control_test.go [hunk 1]
```diff
@@ -24,7 +24,6 @@ type TrafficControl struct {
 	ASNs                                              []tc.ASN                                `json:"asns"`
 	CDNs                                              []tc.CDN                                `json:"cdns"`
 	CacheGroups                                       []tc.CacheGroupNullable                 `json:"cachegroups"`
-	CacheGroupParameterRequests                       []tc.CacheGroupParameterRequest         `json:"cachegroupParameters"`
```
Comment (ocket8888) [inline traffic_ops/testing/api/v4/traffic_control_test.go:27]: Should these also be removed from the fixtures data?

**Review by ocket8888 — COMMENTED** [review]
Docs have links to deleted pages:
```
docs/source/api/migrating-from-v1.rst:96: WARNING: undefined label: to-api-cachegroupparameters
clients/python/trafficops/tosession.py:docstring of trafficops.tosession.TOSession.delete_cache_group_parameters:1: WARNING: undefined label: to-api-cachegroupparameters-id-parameterid
clients/python/trafficops/tosession.py:docstring of trafficops.tosession.TOSession.get_all_cachegroup_parameters:1: WARNING: undefined label: to-api-cachegroupparameters
clients/python/trafficops/tosession.py:docstring of trafficops.tosession.TOSession.get_cachegroup_parameters:1: WARNING: undefined label: to-api-cachegroups-id-parameters
```

There're also like 100 warnings linking to the now-deleted long description 2 and 3 field descriptions in the DS overview - you don't need to fix those, but I thought I'd mention it in case you want to.

File: traffic_ops/testing/api/v4/traffic_control_test.go (hunk 1 above)
Comment (rawlinp) [reply traffic_ops/testing/api/v4/traffic_control_test.go:27]: So it turns out, this didn't even have any fixtures data to remove.

**Comment by rawlinp** [comment]
Docs warnings should all be fixed now.

**Comment by ocket8888** [comment]
Docs warnings are gone, and everything seems to be working properly, but in the past deprecated routes have had warning-level deprecation alerts added to their responses, and `.. deprecated` was placed on the routes in older API versions' docs - do we want to do that here? Or why not?

**Comment by rawlinp** [comment]
I didn't add deprecation notices to the v1 docs since those are supposed to be getting removed soon. Also this reminded me that there are probably some things in Traffic Portal I need to remove as well, so please hold off merging until I've handled TP.

**Comment by ocket8888** [comment]
Well v2 and v3 still have cachegroupparameters stuff in them

Sure, I'll add the WIP label, in case that matters.

**Comment by rawlinp** [comment]
> Well v2 and v3 still have cachegroupparameters stuff in them

Yeah, when I said "I didn't add deprecation notices to the v1 docs", what I meant was "I just added deprecation notices to the v2 and v3 docs." Sorry if that wasn't very clear.

**Comment by ocket8888** [comment]
Oh, I see. Yeah, that's fine, they are slated for removal soon. I've been adding/changing things in the v1 docs but at this point we're definitely sure enough they're going away that we don't need to do that.

**Comment by rawlinp** [comment]
Ok, this is ready for review once again.

**Comment by rawlinp** [comment]
Looks like the `DeprecatedReadHandler` might not handle IMS requests properly...taking a look now.

**Review by ocket8888 — APPROVED** [review]
This all appears to be working correctly and the docs and behavior match. TP changes look good as far as I can tell. Just need that IMS thing to pass.

**Pull request merged**